core/src/test/scala/org/apache/spark/sql/delta/DeltaDDLSuite.scala

```diff
@@ -476,4 +475,40 @@ abstract class DeltaDDLTestBase extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("block table creation with ambiguous delta paths") {
```

Comment:
nit: maybe you could put those 2 tests into one block:
```
Seq(true, false).foreach { enabled =>
  withSQLConf(DeltaSQLConf.DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS.key -> enabled.toString)
```
And then a separate one liner test to check DeltaSQLConf.DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS.key is set to false by default.
```
assert(!spark.conf.get(DeltaSQLConf.DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS))
```